File: apps/backend/app/api/routes/projects.py
from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.schemas.project import ProjectCreate, ProjectRead
from app.services.project_service import ProjectService

router = APIRouter(prefix="/projects", tags=["projects"])


@router.get("", response_model=list[ProjectRead])
def list_projects(db: Session = Depends(get_db)) -> list[ProjectRead]:
    service = ProjectService(db)
    return service.list_projects()


@router.post("", response_model=ProjectRead, status_code=status.HTTP_201_CREATED)
def create_project(payload: ProjectCreate, db: Session = Depends(get_db)) -> ProjectRead:
    service = ProjectService(db)
    return service.create_project(payload)

File: apps/frontend/src/services/projects.ts
import { http } from '@/services/http'
import type { ProjectCreate, ProjectRead } from '@/schemas/project'

export function fetchProjects() {
  return http<ProjectRead[]>('/api/projects')
}

export function createProject(payload: ProjectCreate) {
  return http<ProjectRead>('/api/projects', {
    method: 'POST',
    body: JSON.stringify(payload),
  })
}
